recipes/logr/0.2.0/conanfile.py

```diff
@@ -0,0 +1,105 @@
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration
+import os
+
+class LogrConan(ConanFile):
+    name = "logr"
+    license = "BSD 3-Clause License"
+    homepage = "https://github.com/ngrodzitski/logr"
+    url = "https://github.com/conan-io/conan-center-index"
+    description = "Logger frontend substitution for spdlog, glog, etc for server/desktop applications"
+    topics = ("logger", "development", "util", "utils")
+    generators = "cmake"
+    settings = "os", "compiler", "build_type", "arch"
+    no_copy_source = True
+
+    options = { "backend": ["spdlog", "glog", "log4cplus", "log4cplus-unicode", None] }
+    default_options = { "backend": "spdlog"}
+
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def requirements(self):
+        self.requires("fmt/7.1.2")
+
+        if self.options.backend == "spdlog":
+            self.requires("spdlog/1.8.2")
+        elif self.options.backend == "glog":
+            self.requires("glog/0.4.0")
+        elif self.options.backend == "log4cplus":
+            self.requires("log4cplus/2.0.5")
+        elif self.options.backend == "log4cplus-unicode":
+            self.requires("log4cplus/2.0.5")
```

Comment:
I don't understand the difference between the options `log4cplus` and `log4cplus-unicode`.
I guess if people want unicode, then they can do `-o log4cplus:unicode=True` themselve?

Reply:
Are you suggesting to leave only `log4cplus` (hence removing build method which is all about checking logr-backend-option and log4cplus-unicode-option compatibility)?

Reply:
Yes, I guess your `logr` library will automatically detect whether `log4cplus` is configured with/without unicode?
How does it discriminate between log4cplus with/without unicode?

Also, `build` is a bad place for such tests as it is only checked when creating the conan package.
The latest conan release introduced `validate` for this.

Reply:
log4cplus defines its own `tchar` and logr works with it. So if in logr-package options there is no explicit notion of what alternative of log4clus is used no check is possible, because log4cplus unicode option is the only point which is not required to be in sync with anything.

Reply:
So I think you can remove the `-unicode` option and check my other comment about the `header_only` in `package_id`.